aonvif/client.py

```diff
@@ -274,9 +274,42 @@ def __init__(
 
         # Currently initialized services
         self._services = {}
+        self._capabilities = None
 
         self.to_dict = ONVIFService.to_dict
 
+    def set_capabilities(self, capabilities: dict):
+        """Set custom capabilities.
+
+        :param capabilities: Dictionary of camera capabilities.
+                             It must have the following format:
+                             {
+                                "Media": {
+                                    "XAddr": "http://{host}/{xaddr_path_1},
+                                },
+                                "PTZ": {
+                                    "XAddr": "http://{host}/{xaddr_path_2},
+                                },
+                             }
+        """
+
+        self._validate_capabilities(capabilities)
+        self._capabilities = capabilities
+
+    def _validate_capabilities(self, capabilities: dict):
+        if not isinstance(capabilities, dict):
+            raise RuntimeError('Capabilities type must be dictionary')
+
+        for key, capability in capabilities.items():
+            if not isinstance(key, str):
+                raise RuntimeError('Capabilities key type must be string')
+
+            if not isinstance(capability, dict):
+                raise RuntimeError('Capability type must be dictionary')
+
+            if not isinstance(capability.get('XAddr'), str):
```

Comment:
If we only support `XAddr` key probably its presence should be explicitly verfied, the current message is sort of misleading, if `XAddr` was not provided at all.

Reply:
Fixed